docs/references/file_provider_markdown.md

```diff
@@ -0,0 +1,272 @@
+-   `type`: The type of each piece of data in that field.
+
+`type` must correspond to 1 of the values in the `FieldType` enum. See
+the example below:
+
+```xml
+<RecordInfo>
+  <Field name="Name" size="254" type="v_string" />
+  <Field name="Value" size="254" type="double" scale="2" description="the value" source="internet"/>
+</RecordInfo>
+```
+
+### Record File
+
+The record CSV file contains all of the record information. It looks
+like a CSV file that Designer might use as input. The first row should
+contain field names corresponding to the name of each field in the
+metadata file. The rest of the rows should contain the record data. See
+the example below:
+
+```cs
+Name,Value
+Ten,10
+Twenty,20
+Thirty,30
+```
+
+Running the File Provider
+-------------------------
+
+The file provider is run from a command-line argument. The only argument
+the user must pass in is a path to a JSON file with this format:
+
+1.  `tool`: This is a dictionary with 2 keys: `plugin` and `path`.
+
+     -   `plugin`: Contains a string that corresponds to the class
+         plugin name.
+     -   `path`: Contains a string that corresponds to the folder that
+         contains the plugin, in other words,
+         `{"plugin": "Example", "path": "path/to/ExampleFolder"}`.
+
+2.  `tool\_config`: This is a string that corresponds to the absolute
+    path of the tool\'s configuration file, in other words,
+    `path/to/ExampleToolConfig.xml`.
+3.  `workflow\_config`: This is a string that corresponds to the
+    absolute path of the workflow\'s configuration file, in other words,
+    `path/to/ExampleWorkflowConfig.xml`.
+4.  `inputs`: This is a list of dictionaries that contain information
+    about the tool\'s input connections. It is optional since not all
+    tools have input. The user should specify input information for each
+    input connection that is associated with this tool. Each input
+    dictionary has 3 keys: `anchor_name`, `records`, and `metadata`.
+
+     -   `anchor_name`: Contains a string that corresponds to the name
+         of the input connection\'s anchor, which has to match an input
+         anchor name from the associated configuration file.
+     -   `records`: Contains a string that corresponds to the absolute
+         path of the input connection\'s record file.
+     -   `metadata`: Contains a string that corresponds to the absolute
+         path of the input connection\'s metadata file, in other words,
+         `{"anchor_name": "AnchorName", "records": "path/to/InputRecords.csv", "metadata": "path/to/InputMetadata.xml"}`.
+
+5.  `outputs`: This is a list of dictionaries that specify where the
+    tool\'s output anchor information should be stored. It is optional
+    since not all tools have output. The user should specify output
+    information for each output anchor that is associated with this
+    tool. Each output dictionary has 3 keys: `anchor_name`, `records`,
+    and `metadata`.
```

Comment:
TW review:

`outputs`: This is a list of dictionaries that specify where the
    tool\'s output anchor information should be stored. It's optional
    since not all tools have output. You should specify output
    information for each output anchor that is associated with this
    tool. Each output dictionary has 3 keys: `anchor_name`, `records`,
    and `metadata`.

Reply:
fixed